relentless/simulate/generic.py

```diff
@@ -0,0 +1,301 @@
+            except AttributeError:
+                raise TypeError('{}.{}.{} operation not found.'.format(backend.__module__,backend.__name__,op_name))
+
+            self._op = BackendOp(*self.args,**self.kwargs)
+            self._backend = sim.backend
+
+        return self._op(sim)
+
+    @classmethod
+    def add_backend(cls, backend, module=None):
+        """
+        """
+        # try to deduce module from backend class
+        if module is None:
+            module = backend.__module__
+
+        # setup module (if not already a module)
+        if not inspect.ismodule(module):
+            module = importlib.import_module(module)
+
+        cls.backends[backend] = module
+
+## initializers
+class InitializeFromFile(SimulationOperationAdapter):
+    """Initializes a simulation box and pair potentials from a file.
+
+    Parameters
+    ----------
+    filename : str
+        The file from which to read the system data.
+    options : kwargs
+        Options for file reading.
+    r_buff : float
+        Buffer width (defaults to 0.4).
+
+    """
+    def __init__(self, filename, r_buff=0.4, **options):
+        super().__init__(filename, r_buff, **options)
+
+class InitializeRandomly(SimulationOperationAdapter):
+    """Initializes a randomly generated simulation box and pair potentials.
+
+    Parameters
+    ----------
+    seed : int
+        The seed to randomly initialize the particle locations (defaults to `None`).
+    r_buff : float
+        Buffer width (defaults to 0.4).
+
+    """
+    def __init__(self, seed=None, r_buff=0.4):
+        super().__init__(seed, r_buff)
+
+## integrators
+class MinimizeEnergy(SimulationOperationAdapter):
+    """:py:class:`SimulationOperation` that runs a FIRE energy minimzation until converged.
+
+    Parameters
+    ----------
+    energy_tolerance : float
+        Energy convergence criterion.
+    force_tolerance : float
+        Force convergence criterion.
+    max_iterations : int
+        Maximum number of iterations to run the minimization.
+    dt : float
+        Maximum step size.
+
+    """
+    def __init__(self, energy_tolerance, force_tolerance, max_iterations, dt):
+        super().__init__(energy_tolerance, force_tolerance, max_iterations, dt)
+
+class AddMDIntegrator(SimulationOperationAdapter):
+    """:py:class:`SimulationOperation` to add an integrator (for equations of motion) to the simulation.
+
+    Parameters
+    ----------
+    dt : float
+        Time step size for each simulation iteration.
+
+    """
+    def __init__(self, dt):
+        super().__init__(dt)
+
+class RemoveMDIntegrator(SimulationOperationAdapter):
+    """:py:class:`SimulationOperation` that removes a specified integration operation.
+
+    Parameters
+    ----------
+    add_op : :py:class:`SimulationOperation`
+        The addition/integration operation to be removed.
+
+    """
+    def __init__(self, add_op):
+        super().__init__(add_op)
```

Comment:
These are base classes (they user will not initialize directly), so they don't need to be included here.